firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarState.swift

```diff
@@ -65,39 +65,46 @@ struct ToolbarState: ScreenState, Equatable {
             ToolbarActionType.backButtonStateChanged,
             ToolbarActionType.forwardButtonStateChanged,
             ToolbarActionType.scrollOffsetChanged:
-            guard let toolbarAction = action as? ToolbarAction else { return state }
-            return ToolbarState(
-                windowUUID: state.windowUUID,
-                toolbarPosition: toolbarAction.toolbarPosition ?? state.toolbarPosition,
-                isPrivateMode: state.isPrivateMode,
-                addressToolbar: AddressBarState.reducer(state.addressToolbar, toolbarAction),
-                navigationToolbar: NavigationBarState.reducer(state.navigationToolbar, toolbarAction))
+            return handleToolbarAction(state: state, action: action)
 
         case GeneralBrowserActionType.updateSelectedTab:
-            guard let action = action as? GeneralBrowserAction else { return state }
-            return ToolbarState(
-                windowUUID: state.windowUUID,
-                toolbarPosition: state.toolbarPosition,
-                isPrivateMode: action.isPrivateBrowsing ?? state.isPrivateMode,
-                addressToolbar: AddressBarState.reducer(state.addressToolbar, action),
-                navigationToolbar: NavigationBarState.reducer(state.navigationToolbar, action))
+            return handleGeneralBrowserAction(state: state, action: action)
 
         case ToolbarActionType.toolbarPositionChanged:
-            guard let position = (action as? ToolbarAction)?.toolbarPosition else { return state }
-            return ToolbarState(
-                windowUUID: state.windowUUID,
-                toolbarPosition: position,
-                isPrivateMode: state.isPrivateMode,
-                addressToolbar: AddressBarState.reducer(state.addressToolbar, action),
-                navigationToolbar: NavigationBarState.reducer(state.navigationToolbar, action))
+            return handleToolbarPositionChanged(state: state, action: action)
 
         default:
-            return ToolbarState(
-                windowUUID: state.windowUUID,
-                toolbarPosition: state.toolbarPosition,
-                isPrivateMode: state.isPrivateMode,
-                addressToolbar: state.addressToolbar,
-                navigationToolbar: state.navigationToolbar)
+            return state
```

Comment:
This is another instance where the `struct` was recreated with the same values. I returned it directly since the result is identical. Does anyone have a different perspective?